src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedByKey: GroupsMap<T> = {};
+  const allKeys = items
+    .map((item) => item[key])
+    .filter((item, index, arr) => arr.indexOf(item) === index);
+
+  for (const currentKey of allKeys) {
+    groupedByKey[String(currentKey)] = items.filter(
```

Comment:
Casting the 'currentKey' to a string might not be necessary if the key is already a string. Ensure that the key property is always a string or handle different types accordingly.